test/SlowTests/Issues/RavenDB_15521.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System.IO;
+using System.Text;
+using FastTests;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace SlowTests.Issues
+{
+    public class RavenDB_15521 : RavenTestBase
+    {
+        public RavenDB_15521(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        [Fact]
+        public void ShouldWork()
+        {
+            using (var store = GetDocumentStore())
+            {
+                using (var session = store.OpenSession())
+                {
+                    SimpleDoc doc = new SimpleDoc() { Id = "TestDoc", Name = "State1" };
+                    session.Store(doc);
+
+                    string attachment = "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa";
+                    var stream = new MemoryStream(Encoding.UTF8.GetBytes(attachment));
+                    session.Advanced.Attachments.Store(doc, "TestAttachment", stream);
+
+                    session.SaveChanges();
+                    var changeVector1 = session.Advanced.GetChangeVectorFor(doc);
+                    session.Advanced.Refresh(doc);
```

Comment:
Not sure that this is a good idea.
What happened if there were _other_ changes to the document in the meantime?

Reply:
What do you mean? Refresh is fetching data as it is from the Server.